src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -173,46 +173,57 @@ public void postInject() {
+                    && deviceConfiguration.provisionInfoNodeChanged(node, false)) {
                 try {
-                    // Not using isDeviceConfiguredToTalkToCloud() in order to provide the detailed error
-                    // message to user
-                    deviceConfiguration.validate();
-
-                    Topics configurationTopics = deviceConfiguration.getStatusConfigurationTopics();
-                        configurationTopics.lookup(FLEET_STATUS_PERIODIC_PUBLISH_INTERVAL_SEC)
-                            .dflt(DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC)
-                                .subscribe((why, newv) -> {
-                        int newPeriodicUpdateIntervalSec = Coerce.toInt(newv);
-                        // Do not update the scheduled interval if it is less than the default.
-                        if (newPeriodicUpdateIntervalSec < DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC) {
-                            return;
-                        }
-                        this.periodicPublishIntervalSec = TestFeatureParameters.retrieveWithDefault(Double.class,
-                                FLEET_STATUS_TEST_PERIODIC_UPDATE_INTERVAL_SEC, newPeriodicUpdateIntervalSec)
-                                .intValue();
-                        if (periodicUpdateFuture != null) {
-                            schedulePeriodicFleetStatusDataUpdate(false);
-                        }
-                    });
-
-                    config.getContext().addGlobalStateChangeListener(this::handleServiceStateChange);
-
-                    this.deploymentStatusKeeper.registerDeploymentStatusConsumer(IOT_JOBS,
-                        this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-                    this.deploymentStatusKeeper.registerDeploymentStatusConsumer(LOCAL,
-                        this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-                    this.deploymentStatusKeeper.registerDeploymentStatusConsumer(SHADOW,
-                        this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-                    schedulePeriodicFleetStatusDataUpdate(false);
-
+                    setUpFSS();
                 } catch (DeviceConfigurationException e) {
                     logger.atWarn().kv("errorMessage", e.getMessage()).log(DEVICE_OFFLINE_MESSAGE);
                 }
-                this.isFSSSetupComplete.set(true);
             }
         });
+        try {
+            setUpFSS();
+        } catch (DeviceConfigurationException e) {
+            logger.atWarn().kv("errorMessage", e.getMessage()).log(DEVICE_OFFLINE_MESSAGE);
+        }
+    }
+
+    private void setUpFSS() throws DeviceConfigurationException {
+        // Not using isDeviceConfiguredToTalkToCloud() in order to provide the detailed error message to user
+        deviceConfiguration.validate();
+        if (!isFSSSetupComplete.get()) {
+            initialFSSsetup();
+        }
```

Comment:
The way the atomic boolean is used here is susceptible to a time of check/time of set bug. You should use:
```
if (!isFSSSetupComplete.checkAndSet(false, true)) {
  initialFSSsetup();
}
```
You could then move the contents of `initialFSSsetup` into if check here instead - that way there's no separate method modifying the atomic bool.

Super nit: you are inconsistent with capitalization for `isFSSSetup` and `initialFSSsetup`

Reply:
Do you mean compareAndSet ? I don't see a checkAndSet method in the AtomicBoolean documentation.
I'll make the change as suggested.

Will also make the capitalization consistent.